src/engraving/libmscore/beam.cpp

```diff
@@ -267,17 +267,17 @@ bool Beam::isNoSlope() const
 
 void Beam::alignBeamPosition()
 {
-    BeamPos currentBeamPos = beamPos();
+    PointF currentBeamPos = beamPos();
```

Comment:
As this is not really about x and y coordinates of one of a beams handles. but more about the left and right y-psotions, the `QPair` approach seems to make more sense, doesn't it?

Reply:
`(Q)Point` was the hack here, not QPair`

Reply:
I agree. I will make it separate PR. There you don't just need to change the value type, but probably introduce a new type in `P_TYPE`, because now `Pid::BEAM_POS` is `P_TYPE::POINT`. 
https://github.com/musescore/MuseScore/blob/master/src/engraving/libmscore/property.cpp#L148